pipelines/src/pipelines/configs/xgboost.py

```diff
@@ -0,0 +1,36 @@
+# training & prediction pipeline
+table_prefix = "xgboost_"
```

Comment:
is the intent to provide users with an easy way to provide their own training config (i.e. a platform) or to show users how to run a pipeline (i.e. an example)? The former suggests slightly more complex code which makes usage easier, but modification harder. The latter suggests simple code which is easier to modify, but less useful "as is".

If it's the former I'd suggest these could become dataclasses to make required variables more obvious, if it;s the latter, suggest eliminating the config injection and stick to one method of training. But really it's a question of "vision" for the repository moving forward.

Reply:
My view is that we are mainly aiming for option #2 - this repo serves as an example of how to use Vertex pipelines for production MLOps. We should avoid unnecessary abstraction as much as possible

Reply:
I agree, but then I'd suggest it may be worth considering dropping either Tensorflow or XGBoost so there is only one training method. As it sits currently having both introduces a complication to the codebase that is not necessary for the overall project purpose (fewer moving parts means it's easier for them to understand what's going on).

Eliminating one of the training methods
1. simplifies the pipelines 2. eliminates the need for a number of configuration values 3. makes the purpose of the project easier for a layman to understand (only show what they need to know - running something on vertex).

I don't feel the training methods are sufficiently different to both be needed as examples. It makes sense to me to have both if the example code is of a platform in production, but not if the example code is a _model_ running in production (as a single model would not usually need to have the ability to switch libraries baked in).